neurom/core/tests/test_iter_positions.py

```diff
@@ -0,0 +1,31 @@
+import os
+import numpy as np
+import neurom as nm
+
+_path = os.path.dirname(os.path.abspath(__file__))
+TEST_DATA_PATH = os.path.join(_path, "../../../test_data/")
+
+
+def test_iter_positions():
```

Comment:
Could a couple more test cases be added?

For instance, when loading the `test_data/swc/simple.swc`, which has structure:
```
          (0, 5)
(-5, 5)----- ------ (6, 5)
            |
            |
            |
            | Type = 3
            |
            o origin
            |
            | Type = 2
            |
            |
-5, -4)----- ------ (6, -4)
          (0, -4)
```

I get the following:
```
list(nm.iter_positions(morph, None, 10))                   
[array([-5.,  5.,  0.]), array([5., 5., 0.]), array([ 6., -4.,  0.])]
```

Which seems to skip the the `(6, 5)` endpoint - is this the expected behavior?

With the same morphology, when I do:
`list(nm.iter_positions(morph, None, 1))`, I don't seem to get the endpoints: `[(-5, 5), (6, 5), (-5, -4), (6, -4)]`, which seems counter-intuitive to me.

Reply:
the first is expected - the points returned occur every 10um and there was one at (5, 5), only 1um away

The second does seem strange.

Reply:
> the first is expected - the points returned occur every 10um and there was one at (5, 5), only 1um away

Ah, yes, my bad. I was expecting to have 4 points, but since I chose 10, I didn't get the (-5, -4) one, and got confused - sorry.

`list(nm.iter_positions(morph, None, 9))` gives what I expect: `[array([-4.,  5.,  0.]), array([4., 5., 0.]), array([ 5., -4.,  0.]), array([-5., -4.,  0.])]`

Reply:
I've identified the issue. When iterating across a segment points are interpolated up to (but not including) the end of the segment. when the distance to interpolate across is a multuple of the sample_distance, this means that that point is not sampled if the segment has no children (and is sampled multiple times if it has multiple children). I'll use this morphology for a test case and fix this.

Thanks for spotting this